apps/jobs/views.py

```diff
@@ -141,6 +132,47 @@ def get_number_of_applicants(self, serialized_data):
 
         return serialized_data
 
+    @staticmethod
+    def get_active_jobs_count(serialized_company_data):
+        """
+        Add a new field called "Active Jobs" to the serialized data,
+        that contains the count of active jobs present in the company.
+        """
+
+        for company in serialized_company_data.data:
+            jobs_belong_to_company = Job.objects.filter(
+                company_id=company.get("company_id")
+            )
+            active_jobs = sum(1 for job in jobs_belong_to_company if job.is_active)
+            company.update({"Active Jobs": active_jobs})
+
+        return serialized_company_data
+
+    @action(detail=False, methods=["get"], url_path="details")
+    def retrieve_job_by_id(self, request):
```

Comment:
It is accessible at the endpoint ``jobs/details`` using a GET request. It retrieves the job_id from the query parameter and also validates the job_id to ensure it is a valid UUID. 

> Attempts to retrieve the job data from the database using the Job model.
If the job with the specified job_id exists, it serializes the job data and returns it in the response.
If the job does not exist, it returns a 404 Not Found response.